rust/src/dns/detect.rs

```diff
@@ -114,9 +115,37 @@ pub unsafe extern "C" fn rs_dns_detect_opcode_free(ptr: *mut c_void) {
     }
 }
 
+/// Perform the DNS rcode match.
+///
+/// 1 will be returned on match, otherwise 0 will be returned.
+#[no_mangle]
+pub extern "C" fn rs_dns_rcode_match(
+    tx: &mut DNSTransaction, detect: &mut DetectUintData<u8>, flags: u8,
+) -> u8 {
+    let header_flags = if flags & Direction::ToServer as u8 != 0 {
+        if let Some(request) = &tx.request {
+            request.header.flags
+        } else {
+            return 0;
+        }
+    } else if let Some(response) = &tx.response {
+        response.header.flags
+    } else {
+        return 0;
+    };
```

Comment:
When I ran the tests mentioned in the description of the PR they all came back successful. This is the a major change I made from the last PR. Please let me know if this is correct or if it is causing the check failures?

Reply:
The test that is failing is `requires` https://github.com/OISF/suricata/actions/runs/7655557469/job/20861980818#step:19:85
As you pointed out in a chat, I think that the failures are due to the fact that your SV PR has tests that are outdated, since your Suri PR is current to Suri master.
My guess is that if you rebase and submit a new SV PR, ci checks will go green again :)

Reply:
To answer your question directly: overall, this looks correct to me :)